ark/mibi/tiling_utils_test.py

```diff
@@ -0,0 +1,550 @@
+        # due to the nature of randomization, this test will fail once in a blue moon
+        assert random_center_points_run[11:] != sorted_center_points_run[11:]
+
+        # test 10: randomization for one fov, additional moly point interval,
+        # no moly points between runs
+        sample_tiling_params['moly_run'] = 0
+        tiled_regions_random_some_moly_int_no_run = tiling_utils.create_tiled_regions(
+            sample_tiling_params, sample_moly_point
+        )
+
+        # get the center points created
+        random_center_points_no_run = [
+            (fov['centerPointMicrons']['x'], fov['centerPointMicrons']['y'])
+            for fov in tiled_regions_random_some_moly_int_no_run['fovs']
+        ]
+
+        # define the moly point indices, applies for both sorted and random tiled regions
+        moly_indices_no_run = [3, 6, 8, 12, 15]
+
+        # assert that each moly index is the same in both sorted and random tiled regions
+        for mi in moly_indices_run:
+            random_center_points_run[mi] == sorted_center_points_run[mi]
+
+        # ensure the fov 1 center points are the same for both sorted and random
+        assert random_center_points_no_run[:10] == sorted_center_points_no_run[:10]
+
+        # ensure the random center points for fov 2 contain the same elements as its sorted version
+        misc_utils.verify_same_elements(
+            random_center_points=random_center_points_no_run[10:],
+            sorted_center_points=sorted_center_points_no_run[10:]
+        )
+
+        # test 11: randomization for both fovs, additional moly point interval,
+        # moly points between runs
+        sample_tiling_params['randomize'] = [1, 1]
+        sample_tiling_params['moly_run'] = 1
+        tiled_regions_random_all_moly_int_run = tiling_utils.create_tiled_regions(
+            sample_tiling_params, sample_moly_point
+        )
+
+        # get the center points created
+        random_center_points_run = [
+            (fov['centerPointMicrons']['x'], fov['centerPointMicrons']['y'])
+            for fov in tiled_regions_random_all_moly_int_run['fovs']
+        ]
+
+        # assert that each moly index is the same in both sorted and random tiled regions
+        for mi in moly_indices_run:
+            random_center_points_run[mi] == sorted_center_points_run[mi]
+
+        # make sure the random center points contain the same elements as the sorted center points
+        misc_utils.verify_same_elements(
+            random_center_points=random_center_points_run,
+            sorted_center_points=sorted_center_points_run
+        )
+
+        # the random center points should be ordered differently than the sorted center points
+        # due to the nature of randomization, this test will fail once in a blue moon
+        assert random_center_points_run != sorted_center_points_run
+
+        # test 12: randomization for both fovs, additional moly point interval,
```

Comment:
These tests are going to be super hard to maintain. We have 12 copies of almost the exact same code, each testing one of the possible permutations to the inputs. Given that these inputs don't actually interact with each other, I don't think it's necessary to test every possible combination. 

Either way, this is begging for a for loop. We should loop over the possible values of randomization (yes/no), molly point (yes/no), etc. Then, based on the condition, check whether it matches expected. This would collapse all 12 of these tests into a single test.

Reply:
I've definitely been meaning to get to this, the tests are only this way right now to make it super clear what each is testing.

Reply:
Sounds good